Marksman/Parser.fs

```diff
@@ -248,13 +250,30 @@ module Markdown =
 
                 let range = sourceSpanToRange text h.Span
 
+                let heading = {
+                    level = level
+                    isTitle = parserSettings.titleFromHeading && level <= 1
+                    title = Node.mkText title titleRange
+                    disambiguation = None
+                    scope = range
+                }
+
+                let slugAmbiguous = Heading.slug heading
```

Comment:
Let's gate this under a config. This function already accepts parser settings. You could defined a new config var in `Config`, thread it into `ParserSettings`, and check here.